drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1752,12 +1686,17 @@ static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6])
 }
 #endif
 
-static void generate_mac(uint8_t mac_addr[6])
+static void generate_mac(uint8_t mac_addr[6], const struct eth_sam_dev_cfg *const cfg)
 {
-#if DT_INST_NODE_HAS_PROP(0, mac_eeprom)
-	get_mac_addr_from_i2c_eeprom(mac_addr);
-#elif DT_INST_PROP(0, zephyr_random_mac_address)
-	gen_random_mac(mac_addr, ATMEL_OUI_B0, ATMEL_OUI_B1, ATMEL_OUI_B2);
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(mac_eeprom)
+	if (cfg->mac_addr_src == MAC_ADDR_SOURCE_EEPROM) {
+		get_mac_addr_from_i2c_eeprom(mac_addr, cfg->mac_eeprom);
```

Comment:
we only support this method if only one instance is used, therefore we don't need to save the `const struct i2c_dt_spec` in config and can still use the old code for that.